rocAL/source/readers/image/file_source_reader.cpp

```diff
@@ -140,9 +150,51 @@ void FileSourceReader::reset() {
     if (_shuffle) std::random_shuffle(_file_names.begin(), _file_names.end());
     _read_counter = 0;
     _curr_file_idx = 0;
+    if (_stick_to_shard == true) {
+        _curr_file_idx = 0;
+        if (_shard_size > 0) {
+            // Reset the variables
+            _last_batch_padded_size = _read_counter = _curr_file_idx = 0;
+            int read_start_index = 0;
+            if (_shard_size < _batch_count)
+                read_start_index = _batch_count;
+            else {
+                if (_shard_size % _batch_count)
+                    read_start_index = _shard_size;
+                else
+                    read_start_index = _shard_size + (_shard_size % _batch_count);
+            }
+            // To re-arrange the starting index of file-loading keeping in mind the shard_size and batch size
+            std::rotate(_file_names.begin(), _file_names.begin() + read_start_index, _file_names.end());
```

Comment:
why std::rotate is needed?
if its just shuffling, std::random_shuffle wont work?